### PR TITLE
feat: introduce form related fields in process index

### LINKED ISSUE #23043: Add Tasklist specific fields to operate-process index

### PR DESCRIPTION
**Pull request by panagiotisgts**
## Description

Introduce the `formId` and the `isPublic` fields over the `operate-process` index along with the respective chagnes to the ProcessHandler

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #23043

### CODE REVIEW

File: webapps-schema/src/main/java/io/camunda/webapps/schema/entities/operate/ProcessEntity.java [hunk 1]
```diff
@@ -71,7 +75,9 @@ public boolean equals(final Object o) {
         && Objects.equals(bpmnXml, that.bpmnXml)
         && Objects.equals(resourceName, that.resourceName)
         && Objects.equals(flowNodes, that.flowNodes)
-        && Objects.equals(tenantId, that.tenantId);
+        && Objects.equals(tenantId, that.tenantId)
+        && Objects.equals(formId, that.formId)
+        && Objects.equals(isPublic, that.isPublic);
```
Comment (nathansandi) [inline webapps-schema/src/main/java/io/camunda/webapps/schema/entities/operate/ProcessEntity.java:80]: Are we ready to come with the public information over here? Or will be implicit for the future? Once it comes related to authentication (not previewed from 8.7 for this topic)

File: webapps-schema/src/main/java/io/camunda/webapps/schema/entities/operate/ProcessEntity.java [hunk 2]
```diff
@@ -168,4 +178,22 @@ public ProcessEntity setTenantId(final String tenantId) {
     this.tenantId = tenantId;
     return this;
   }
+
+  public String getFormKey() {
```
Comment (nathansandi) [inline webapps-schema/src/main/java/io/camunda/webapps/schema/entities/operate/ProcessEntity.java:182]: 🔧  I think here would be getFormId, correct?

**Review by nathansandi — CHANGES_REQUESTED** [review]
Small fix on get method,

also just related to isPublic, I think it is fine to add it already, but have a remark this wont be supported (as far as I know) in 8.7. But yes, it will be supported in further versions.

File: webapps-schema/src/main/java/io/camunda/webapps/schema/entities/operate/ProcessEntity.java (hunk 1 above)
Comment (panagiotisgts) [reply webapps-schema/src/main/java/io/camunda/webapps/schema/entities/operate/ProcessEntity.java:80]: This is just information extracted from the BPMN file, as it was before over on the `tasklist-process` index. I believe accessing public processes will be bound to the authentication changes so as to actually enable this field.

**Review by nathansandi — APPROVED** [review]
Looks good from my side.

**Pull request merged**